nhais-adaptor/edifact/models/message.py

```diff
@@ -0,0 +1,155 @@
+    takes in specific values required to generate a message trailer
+    example: UNT+18+00000003'
+    """
+
+    SEGMENT_KEY = "UNT"
+
+    def __init__(self, number_of_segments, sequence_number):
+        """
+        :param number_of_segments: the total number of segments in the message including the header and trailer
+        :param sequence_number: a unique reference of the message
+        """
+        segment_value = f"{number_of_segments}+{sequence_number}"
+        super().__init__(key=self.SEGMENT_KEY, value=segment_value)
+
+
+class MessageBeginning(SegmentCollection):
+    """
+    A collection of segments that represent the start of a message
+    Provides details regarding the purpose of the message
+    """
+
+    def __init__(self, party_id, date_time, ref_number):
+        """
+        Create the collection of segments represent the beginning of a message
+        :param party_id: Cipher of the NHAIS system, can be 2 or 3 characters
+        :param date_time: the date time stamp of the message
+        :param ref_number: a reference number for registration transaction type
+        """
+        formatted_date_time = datetime.strptime(date_time, '%Y-%m-%d %H:%M:%S.%f').strftime('%Y%m%d%H%M')
+        segments = [
+            Segment(key="BGM", value="++507"),
+            Segment(key="NAD", value=f"FHS+{party_id}:954"),
+            Segment(key="DTM", value=f"137:{formatted_date_time}:203"),
+            Segment(key="RFF", value=f"950:{ref_number}"),
+        ]
+        super().__init__(segments=segments)
+
+
+class MessageSegmentTrigger1(SegmentCollection):
+    """
+    A collection of segments to provide registration information for GP patients.
+    """
+
+    def __init__(self, transaction_number, party_id, acceptance_code, acceptance_type, date_time, location):
+        """
+        :param transaction_number: a unique transaction number. NHAIS will reference this in its response
+        :param party_id: GMC National code and the Local GP Code of the patient's GP (separated by “,”).
+        :param acceptance_code: The acceptance code "A" for Acceptance
+        :param acceptance_type: The acceptance type "1" for a Birth
+        :param date_time: date of the registration
+        :param location: the patients place of birth
+        """
+        formatted_date_time = datetime.strptime(date_time, '%Y-%m-%d %H:%M:%S.%f').strftime('%Y%m%d')
+        segments = [
+            Segment(key="S01", value="1"),
+            Segment(key="RFF", value=f"TN:{transaction_number}"),
+            Segment(key="NAD", value=f"GP+{party_id}:900"),
+            Segment(key="HEA", value=f"ACD+{acceptance_code}:ZZZ"),
+            Segment(key="HEA", value=f"ATP+{acceptance_type}:ZZZ"),
+            Segment(key="DTM", value=f"956:{formatted_date_time}:102"),
+            Segment(key="LOC", value=f"950+{location}"),
+        ]
+        super().__init__(segments=segments)
+
+
+class MessageSegmentTrigger2(SegmentCollection):

```

Comment:
Is there any better name we can use for these two classes?

Reply:
I was thinking about this. Segment Trigger 2 refers to the patient details Segment. But this is always a Segment Trigger 2 to edifact. I suppose that can detail can be abstracted away here. Will update to more meaningful names 👍